testutil/keeper/profile.go

```diff
@@ -2,18 +2,23 @@ package keeper
 
 import (
 	"context"
+	sdk "github.com/cosmos/cosmos-sdk/types"
 	"math/rand"
 
-	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/stretchr/testify/require"
 
 	"github.com/tendermint/spn/testutil/sample"
 	profiletypes "github.com/tendermint/spn/x/profile/types"
 )
 
 // CreateCoordinator creates a coordinator in the store and returns ID with associated address
-func (tm TestMsgServers) CreateCoordinator(ctx context.Context, r *rand.Rand) (id uint64, address sdk.AccAddress) {
-	addr := sample.AccAddress(r)
+func (tm TestMsgServers) CreateCoordinator(ctx context.Context, r *rand.Rand, optionalAddress ...string) (id uint64, address sdk.AccAddress) {
```

Comment:
Optionally, to be more semantic you could also define a:

```go
CreateCoordinatorWithAddr(ctx context.Context, address string)
```

And then change `CreateCoordinator` signature to only receive the context and randomizer, and its body to call `CreateCoordinatorWithAddr`:

```go
tm.CreateCoordinatorWithAddr(ctx, sample.AccAddress(r))
```

These changes would allow to remove the variadic string argument.

Alternatively the logic could be changed to `CreateCoordinator` and `CreateCoordinatorWithRandomAddr`.

Reply:
agreed, that seems a lot cleaner